pkg/kritis/crd/securitypolicy/securitypolicy.go

```diff
@@ -68,9 +86,18 @@ func ValidateImageSecurityPolicy(isp v1beta1.ImageSecurityPolicy, image string,
 		return violations, nil
 	}
 	// Now, check vulnz in the image
-	vulnz, err := client.GetVulnerabilities(image)
-	if err != nil {
-		return nil, err
+	var vulnz []metadata.Vulnerability
+	var err error
+	if _, ok := v.imageToVulnz[image]; ok {
+		glog.Infof("found cached vulnz for image %s", image)
+		vulnz = v.imageToVulnz[image]
+	} else {
+		glog.Infof("no cached vulnz for image %s, getting from api", image)
+		vulnz, err = client.GetVulnerabilities(image)
+		if err != nil {
+			return nil, err
```

Comment:
It was like this before, but now that the function is larger, do you mind annotating the error message here in some way? Such as:

"GetVulnerabilities(%s): %v", image, err
"failed to get vulnerabilities for %s: %v", image, err

I feel like ValidateImageSecurityPolicy does so much, that if a user ever gets an unhelpful error back from it, it's going to be difficult for us to analyze later what the source was. 

Context: https://golang.org/doc/effective_go.html#errors

"When feasible, error strings should identify their origin, such as by having a prefix naming the operation or package that generated the error. "

Reply:
i've update this particular error with more info as suggested.  If tracing error messages becomes an issue, it make make sense to use [pkg/errors](https://github.com/pkg/errors) throughout kritis.